airflow/www/static/js/trigger.js

```diff
@@ -137,6 +137,14 @@ function initForm() {
           });
           field.on("blur", updateJSONconf);
           objectFields.set(elements[i].name, field);
+        } else if (
+          elements[i].attributes.valuetype &&
+          elements[i].attributes.valuetype.value === "array"
+        ) {
+          if (elements[i].value) {
+            elements[i].value = JSON.parse(elements[i].value).join("\n");
```

Comment:
I do not fully understand, why do you parse the original value as JSON and then put the elements bak into a simple list again?

Reply:
This is to handle the input in case of a failed validation. The input is put back into the field but if its not formatted back to an string the textarea renders the raw json array
![image](https://github.com/apache/airflow/assets/36713373/008d51b9-c626-4eb4-aa50-e12bf577db0f)

Reply:
And what is the outcome if the value of the field is filled from a default (e.g. 3 elements) - then JSON parsing will fail?

Reply:
Yes when the form is triggered with the content from the screenshot, it will be handled as a single string element in an array.